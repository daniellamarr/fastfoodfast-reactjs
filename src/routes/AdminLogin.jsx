import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { adminLogin } from '../actions/authActions';
import Loader from '../components/Loader.jsx';

/**
 * Login
 */
export class AdminLogin extends Component {
  /**
   * Constructor
   */
  constructor() {
    super();

    this.state = {
      email: '',
      password: '',
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  /**
   * onChange
   * @param {object} e
   * @returns {null} null
   */
  onChange(e) {
    this.setState({ [e.target.name]: e.target.value });
  }

  /**
   * onSubmit
   * @param {object} e
   * @returns {null} null
   */
  onSubmit(e) {
    e.preventDefault();

    this.props.adminLogin({
      email: this.state.email,
      password: this.state.password,
    });
  }

  /**
   * @returns {JSX} jsx
   */
  render() {
    return (
      <div id="loginPage">
        { this.props.auth.loadingStatus ? <Loader /> : null }
        <section id="login">
          <center>
            <div className="login">
            <h4>This authentication page is reserved only for Admin</h4>
              <form id="adminLoginForm" method="post" onSubmit={this.onSubmit}>
                <input
                  type="email"
                  name="email"
                  id="email"
                  placeholder="Email"
                  onChange={this.onChange}
                  required
                />
                <input
                  type="password"
                  name="password"
                  id="password"
                  placeholder="Password"
                  onChange={this.onChange}
                  required
                />
                <button type="submit">Sign in </button>
              </form>
            </div>
          </center>
        </section>
        <div className="error hide"></div>
      </div>
    );
  }
}

AdminLogin.propTypes = {
  auth: PropTypes.shape({
    loadingStatus: PropTypes.bool.isRequired,
  }),
  adminLogin: PropTypes.func.isRequired,
};

export const mapStateToProps = state => ({
  auth: state.auth,
});

export default connect(mapStateToProps, { adminLogin })(AdminLogin);
