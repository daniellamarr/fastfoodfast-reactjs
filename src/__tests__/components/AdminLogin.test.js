/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import { mount } from 'enzyme';
import { AdminLogin } from '../../routes/AdminLogin.jsx';

let wrapper;

const props = {
  auth: {
    loadingStatus: false,
  },
  adminLogin: () => {},
};

beforeEach(() => {
  wrapper = mount(<AdminLogin {...props} />);
});

describe('Login Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should find simulate email field change', () => {
    wrapper.find('#email').simulate('change');
    expect(wrapper.state().email).toEqual('');
  });

  it('should find login form submit', () => {
    wrapper.find('#adminLoginForm').simulate('submit', {
      preventDefault: () => {},
    });
    const props2 = { ...props, auth: { loadingStatus: true } };
    wrapper = mount(<AdminLogin {...props2} />);
    expect(wrapper.props().auth.loadingStatus).toEqual(true);
  });
});
