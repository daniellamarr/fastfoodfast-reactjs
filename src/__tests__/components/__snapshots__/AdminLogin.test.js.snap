// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component should match snapshot 1`] = `
<AdminLogin
  adminLogin={[Function]}
  auth={
    Object {
      "loadingStatus": false,
    }
  }
>
  <div
    id="loginPage"
  >
    <section
      id="login"
    >
      <center>
        <div
          className="login"
        >
          <h4>
            This authentication page is reserved only for Admin
          </h4>
          <form
            id="adminLoginForm"
            method="post"
            onSubmit={[Function]}
          >
            <input
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email"
              required={true}
              type="email"
            />
            <input
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required={true}
              type="password"
            />
            <button
              type="submit"
            >
              Sign in 
            </button>
          </form>
        </div>
      </center>
    </section>
    <div
      className="error hide"
    />
  </div>
</AdminLogin>
`;
